<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindowClass</class>
 <widget class="QMainWindow" name="MainWindowClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>680</width>
    <height>372</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>680</width>
    <height>0</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>680</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MODBUS node emulator v0.1a</string>
  </property>
  <property name="toolButtonStyle">
   <enum>Qt::ToolButtonIconOnly</enum>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="1">
     <widget class="QGroupBox" name="groupBox_3">
      <property name="minimumSize">
       <size>
        <width>300</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>300</width>
        <height>16000</height>
       </size>
      </property>
      <property name="title">
       <string>Node settings</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="0" column="1">
        <widget class="QSpinBox" name="nodeAddrSpin">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>240</number>
         </property>
         <property name="value">
          <number>42</number>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QPushButton" name="nodeSettingsSetButton">
         <property name="text">
          <string>Set</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>Node number</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="2" column="0" rowspan="2" colspan="3">
     <widget class="QGroupBox" name="groupBox_2">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>Node data (holding registers only)</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QTableWidget" name="nodeTable">
         <property name="minimumSize">
          <size>
           <width>642</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>642</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="rowCount">
          <number>1</number>
         </property>
         <property name="columnCount">
          <number>8</number>
         </property>
         <attribute name="horizontalHeaderVisible">
          <bool>true</bool>
         </attribute>
         <attribute name="horizontalHeaderCascadingSectionResizes">
          <bool>true</bool>
         </attribute>
         <attribute name="horizontalHeaderDefaultSectionSize">
          <number>80</number>
         </attribute>
         <attribute name="horizontalHeaderStretchLastSection">
          <bool>false</bool>
         </attribute>
         <attribute name="verticalHeaderVisible">
          <bool>false</bool>
         </attribute>
         <row>
          <property name="text">
           <string/>
          </property>
         </row>
         <column>
          <property name="text">
           <string>0</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>1</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>2</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>3</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>4</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>5</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>6</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>7</string>
          </property>
         </column>
         <item row="0" column="0">
          <property name="text">
           <string>3</string>
          </property>
         </item>
         <item row="0" column="1">
          <property name="text">
           <string>1</string>
          </property>
         </item>
         <item row="0" column="2">
          <property name="text">
           <string>1</string>
          </property>
         </item>
         <item row="0" column="3">
          <property name="text">
           <string>3</string>
          </property>
         </item>
         <item row="0" column="4">
          <property name="text">
           <string>7</string>
          </property>
         </item>
         <item row="0" column="5">
          <property name="text">
           <string>6</string>
          </property>
         </item>
         <item row="0" column="6">
          <property name="text">
           <string>6</string>
          </property>
         </item>
         <item row="0" column="7">
          <property name="text">
           <string>6</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QGroupBox" name="groupBox_4">
      <property name="title">
       <string>Node info</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="0" column="0">
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>Packets recieved</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>Packets errors</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLabel" name="lblPacketCnt">
         <property name="text">
          <string>0</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLabel" name="lblPacketErr">
         <property name="text">
          <string>0</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_11">
         <property name="text">
          <string>Last status</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QLabel" name="lblPacketLst">
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_9">
         <property name="text">
          <string>Non-MODBUS</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLabel" name="lblNonMB">
         <property name="text">
          <string>0</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="0" rowspan="2">
     <widget class="QGroupBox" name="groupBox">
      <property name="minimumSize">
       <size>
        <width>300</width>
        <height>186</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>300</width>
        <height>160000</height>
       </size>
      </property>
      <property name="title">
       <string>Port settings</string>
      </property>
      <property name="flat">
       <bool>false</bool>
      </property>
      <property name="checkable">
       <bool>false</bool>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="1" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Baudrate</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Data bits</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Port</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Flow control</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QComboBox" name="bitsSelectCombo">
         <property name="currentIndex">
          <number>3</number>
         </property>
         <item>
          <property name="text">
           <string>5</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>6</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>7</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>8</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>9</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QComboBox" name="pairSelectCombo">
         <item>
          <property name="text">
           <string>None</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Even</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Odd</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Pairity</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QComboBox" name="baudSelectCombo">
         <property name="currentIndex">
          <number>4</number>
         </property>
         <item>
          <property name="text">
           <string>9600</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>19200</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>38400</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>57600</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>115200</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QComboBox" name="flowSelectCombo">
         <item>
          <property name="text">
           <string>None</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Software</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Hardware</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QComboBox" name="portSelectCombo">
         <property name="currentIndex">
          <number>-1</number>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QPushButton" name="portOpenButton">
         <property name="text">
          <string>Open</string>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <widget class="QPushButton" name="portCloseButton">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Close</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="2" rowspan="2">
     <spacer name="horizontalSpacer">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="mainwindow.qrc"/>
 </resources>
 <connections/>
 <slots>
  <slot>portData(QString)</slot>
  <slot>openPort()</slot>
  <slot>closePort()</slot>
 </slots>
</ui>
